src-ts/tools/learn/learn-lib/my-certifications-provider/my-certifications-functions/certificate-progress.decorators.ts

```diff
@@ -0,0 +1,17 @@
+import { LearnMyCertificationProgress } from './learn-my-certification-progress.model'
+
+export function decorateCompletedPercentage(myCertification: LearnMyCertificationProgress): LearnMyCertificationProgress {
```

Comment:
The completion percentage for a particular module comes back in the API response as `completedPercentage`, an integer value. 
```JSON
{
    "currentLesson": "learn-html-by-building-a-cat-photo-app/step-4",
    "provider": "freeCodeCamp",
    "certificationId": "9bd93a8a-1fcb-405a-b2e3-4a283915bbca",
    "courseKey": "2022/responsive-web-design",
    "id": "927abff4-7af9-4145-8ba1-577c16e64e3d",
    "userId": "testflyjets",
    "courseId": "927abff4-7af9-4145-8ba1-577c16e64e2e",
    "startDate": "2022-06-23T04:01:40.000Z",
    "modules": [
        {
            "lessonCount": 67,
            "moduleStatus": "in-progress",
            "completedLessons": [
                {
                    "dashedName": "step-1",
                    "completedDate": "2022-06-23T04:01:40.000Z"
                },
                {
                    "dashedName": "step-2",
                    "completedDate": "2022-06-23T04:01:40.000Z"
                },
                {
                    "dashedName": "step-3",
                    "completedDate": "2022-06-24T04:01:40.000Z"
                }
            ],
            "module": "learn-html-by-building-a-cat-photo-app",
>>          "completedPercentage": 4
        }
    ],
    "certification": "responsive-web-design",
    "status": "in-progress"
}
```
Should we be using that here instead?

Reply:
@vas3a I can easily calculate that in the API call and return it -- I think that makes more sense.

Reply:
@vas3a In your code, are you iterating over all of the modules in the progress record, or over all of the courses in the certification? The progress record currently only contains data for modules the user has started (and I realize there isn't a way to start a new module for a certification you've already started -- I will fix that).

Where in the UI are we showing overall progress towards a certification?

Reply:
Hey @testflyjets !

>I can easily calculate that in the API call and return it -- I think that makes more sense.

sounds good! 

>are you iterating over all of the modules in the progress record, or over all of the courses in the certification

I;m iterating over all modules in the progress record, but it should be over all modules of the certification/course. I didn't realize the progress model doesn't return all modules. 

> Where in the UI are we showing overall progress towards a certification?

In the my learning screen & course details screen we're seeing the overall progress in that progress bar - it's not module specific.